libbeat/common/xml.go

```diff
@@ -0,0 +1,42 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package common
+
+import (
+	"github.com/clbanning/mxj/v2"
+)
+
+// UnmarshalXML takes a slice of bytes, and returns a map[string]interface{}.
+// If the slice is not valid XML, it will return an error.
+func UnmarshalXML(body []byte, prepend bool, toLower bool) (obj map[string]interface{}, err error) {
```

Comment:
Do we need this function somewhere else in the future, or is it specific to the processor being implemented here?

`libbeat/common` is our sin bin for almost everything. Let's try to not grow it if possible. If we need decoding helpers or similar we should introduce appropriate packages.

Reply:
The general idea is to be able to reuse it (if we want to) on other inputs, I added a small section on it in the PR post:

`"It adds a small XML Unmarshal helper function to the common folder in Libbeat, the reason this is added is so that not only processors can utilize it if/when needed, but also inputs. For example I plan to reuse this in http_endpoint to add XML support there as well. Other beats/inputs that would benefit from this is Winlogbeat."`

Reply:
After a discussion with Andrew I will make this a subpackage of the processor instead, which will make it cleaner and more similar to `decode_cef`